### PR TITLE
tidb-binlog: support binlog for clustered index

### PR DESCRIPTION
**Pull request by Liuxiaozhen12**
### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.1 (TiDB 5.1 versions)
- [x] v5.0 (TiDB 5.0 versions) version-specific
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/6405
- Other reference link(s):
- https://github.com/pingcap/tidb/issues/24133
- https://github.com/pingcap/tidb/pull/24135
- https://github.com/pingcap/tidb-binlog/pull/1040

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [x] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by lysu — APPROVED** [review]

File: tidb-binlog/tidb-binlog-overview.md [hunk 1]
```diff
@@ -54,11 +54,16 @@ The TiDB Binlog cluster is composed of Pump and Drainer.
 
 ## Notes
 
-* TiDB Binlog is not compatible with the following features introduced in TiDB v5.0 and they cannot be used together. It is recommended to use [TiCDC](/ticdc/ticdc-overview.md) instead of TiDB Binlog:
+* In v5.1, the incompatibility between introduced clustered index and TiDB Binlog has been solved. After you update TiDB Binlog and TiDB Server to v5.1 and enable TiDB Binlog, TiDB will support creating clustered indexes and data insertion, deletion, and update of the created clustered index tables will be replicated downstream via TiDB Binlog. Pay attention to the replication of clustered index tables:
```
Comment (TomShawn) [inline tidb-binlog/tidb-binlog-overview.md:57]:
```suggestion
* In v5.1, the incompatibility between the clustered index feature introduced in v5.0 and TiDB Binlog has been resolved. After you upgrade TiDB Binlog and TiDB Server to v5.1 and enable TiDB Binlog, TiDB will support creating tables with clustered indexes; data insertion, deletion, and update on the created tables with clustered indexes will be replicated to the downstream via TiDB Binlog. When you use TiDB Binlog to replicate the tables with clustered indexes, pay attention to the following:
```

File: tidb-binlog/tidb-binlog-overview.md [hunk 2]
```diff
@@ -54,11 +54,16 @@ The TiDB Binlog cluster is composed of Pump and Drainer.
...
 
-    - [TiDB Clustered Index](/clustered-indexes.md#limitations): After TiDB Binlog is enabled, TiDB does not allow creating clustered indexes with non-single integer columns as primary keys; data insertion, deletion, and update of the created clustered index tables will not be replicated downstream via TiDB Binlog.
-    - TiDB system variable [tidb_enable_async_commit](/system-variables.md#tidb_enable_async_commit-new-in-v50): After TiDB Binlog is enabled, performance cannot be improved by enabling this option.
-    - TiDB system variable [tidb_enable_1pc](/system-variables.md#tidb_enable_1pc-new-in-v50): After TiDB Binlog is enabled, performance cannot be improved by enabling this option.
+    - If you manually control the upgrade sequence to upgrade v5.0, make sure that TiDB binlog is upgraded to v5.1 before upgrading the TiDB server to v5.1.
```
Comment (TomShawn) [inline tidb-binlog/tidb-binlog-overview.md:59]: @lysu PTAL whether my comment is correct. Thanks!
```suggestion
    - If you have upgraded the cluster to v5.1 from v5.0 by manually controlling the upgrade sequence, make sure that TiDB binlog is upgraded to v5.1 before upgrading the TiDB server to v5.1.
```

File: tidb-binlog/tidb-binlog-overview.md [hunk 3]
```diff
@@ -54,11 +54,16 @@ The TiDB Binlog cluster is composed of Pump and Drainer.
...
+    - It is recommended to configure the system variable [`tidb_enable_clustered_index`](/system-variables.md#tidb_enable_clustered_index-new-in-v50) to a same value to ensure that the structure of TiDB clustered index tables between the upstream and downstream is consistent.
+
+* TiDB Binlog is incompatible with the following features introduced in TiDB v5.0 and they cannot be used together.
+
+    - [TiDB Clustered Index](/clustered-indexes.md#limitations): After TiDB Binlog is enabled, TiDB does not allow creating clustered indexes with non-single integer columns as primary keys; data insertion, deletion, and update of the created clustered index tables will not be replicated downstream via TiDB Binlog. If you need to replicate tables with clustered indexes, update to v5.1 or use [TiCDC](/ticdc/ticdc-overview.md) instead.
```
Comment (TomShawn) [inline tidb-binlog/tidb-binlog-overview.md:64]:
```suggestion
    - [TiDB Clustered Index](/clustered-indexes.md#limitations): After TiDB Binlog is enabled, TiDB does not allow creating clustered indexes with non-single integer columns as primary keys; data insertion, deletion, and update of the created clustered index tables will not be replicated downstream via TiDB Binlog. If you need to replicate tables with clustered indexes, upgrade your cluster to v5.1 or use [TiCDC](/ticdc/ticdc-overview.md) instead.
```

File: tidb-binlog/tidb-binlog-overview.md (hunk 2 above)
Comment (lysu) [reply tidb-binlog/tidb-binlog-overview.md:59]: LGTM

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**